tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestURLs.java

```diff
@@ -0,0 +1,419 @@
+                        if (siteType == SiteType.SITE) {
+                            error.add(Issue.MAKE_RELATIVE);
+                        }
+                    case "unicode.org":
+                        if (file.startsWith("/cldr/data/")
+                                || file.startsWith("unicode.org/cldr/repository_access")
+                                || file.startsWith("/cldr/trac/")) {
+                            error.add(Issue.OLDER_SITE);
+                        }
+                        break;
+                    case "www.unicode.org":
+                        if (file.startsWith("/reports/tr35/")) {
+                            if (siteType == SiteType.SPEC) {
+                                error.add(Issue.MAKE_RELATIVE);
+                            }
+                        } else if (file.startsWith("/repos")) {
+                            error.add(Issue.OLDER_SITE);
+                        }
+                        break;
+                    case "github.com":
+                        if (file.contains("/docs/")) {
+                            error.add(Issue.BAD_GITHUB_DOCS);
+                        }
+                        break;
+                    case "unicode-org.github.io":
+                        if (file.startsWith("/cldr-staging/")) {
+                            error.add(Issue.BAD_GITHUB);
+                        }
+                        break;
+                    case "www.google.com":
+                        if (file.startsWith("/url?") || file.startsWith("/search?")) {
+                            error.add(Issue.SEARCH);
+                        }
+                        break;
+                }
+            }
+        }
+    }
+
+    public static Pattern PROTOCOL_PAT = Pattern.compile("[a-z0-9+.\\-]+:"); //
+
+    public static Pattern OLD_ANCHOR = Pattern.compile(".*#h\\..*"); //
+
+    public static Pattern URL_PAT =
+            Pattern.compile(
+                    "(?:\\bhref=(?:" //
+                            + "[\"]([^\"]*)" //
+                            + "|[']([^']*)))"
+                            + "|\\]\\(([^)]*)"); //
+
+    public static void getUrls(String line, int lineNumber, Multimap<Integer, String> lineToUrls) {
+        Matcher m = URL_PAT.matcher(line);
+        int count = 0;
+        while (m.find()) {
+            for (int i = 1; i <= m.groupCount(); ++i) {
+                String group = m.group(i);
+                if (group != null) {
+                    lineToUrls.put(lineNumber, group);
+                    ++count;
+                    break;
+                }
+            }
+        }
+        // For debugging
+        //        if (count == 0) {
+        //            int start = line.indexOf("href");
+        //            System.out.println(RegexUtilities.showMismatch(url, line.subSequence(start,
+        // line.length())));
+        //            int d=0;
+        //        }
+    }
+
+    public static boolean doesURLExist(URL url) {
```

Comment:
use the `HttpStatusCache` instead, persists.

but, i'd really rather not rewrite an external link checker here, it would be better to use lychee which already deals with this situation.   See https://github.com/unicode-org/cldr/pull/4044 which link checks the spec/rfc - it's only blocked because it would fail until the links were cleaned up.